### PR TITLE
Add RMSLayerNorm Module

### PR DESCRIPTION
**Pull request by BBuf**
libai 需求：添加 RMSLayerNorm。

图片

### CODE REVIEW

File: oneflow/core/functional/impl/math_functor.cpp [hunk 1]
```diff
@@ -26,12 +26,14 @@ limitations under the License.
...
 #include "oneflow/core/functional/function_library.h"
+#include "oneflow/core/functional/functional_api.yaml.h"
 #include "oneflow/core/functional/impl/common.h"
 #include "oneflow/core/functional/impl/unary_functor.h"
 #include "oneflow/core/job/lazy_mode.h"
 #include "oneflow/core/job/sbp_parallel.h"
 #include "oneflow/core/functional/tensor_processor.h"
 
+#include <memory>
 #include <sstream>
```
Comment (zhongshsh) [inline oneflow/core/functional/impl/math_functor.cpp:37]: 新增的 include 是不必要

File: python/oneflow/nn/modules/normalization.py [hunk 2]
```diff
@@ -318,6 +318,49 @@ def extra_repr(self) -> str:
         )
 
 
+class RMSLayerNorm(Module):
+    """
+    Construct a layernorm module in the T5 style. No bias and no subtraction of mean.
+
```
Comment (zhongshsh) [inline python/oneflow/nn/modules/normalization.py:324]: 新增文档后需要添加对应的算子到 rst

File: python/oneflow/test/modules/test_t5_layernorm.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import math
+import numpy as np
+from oneflow.test_utils.test_util import GenArgList
+
+import oneflow as flow
+import oneflow.unittest
+import torch
+
+
+class TorchT5LayerNorm(torch.nn.Module):
```
Comment (zhongshsh) [inline python/oneflow/test/modules/test_t5_layernorm.py:29]: 可以添加 global 测试

**Review by zhongshsh — APPROVED** [review]

File: oneflow/core/functional/impl/math_functor.cpp (hunk 1 above)
Comment (BBuf) [reply oneflow/core/functional/impl/math_functor.cpp:37]: 好的。

File: python/oneflow/test/modules/test_t5_layernorm.py (hunk 3 above)
Comment (BBuf) [reply python/oneflow/test/modules/test_t5_layernorm.py:29]: 这个Module在PyTorch里没有同名的Module，不太好基于autotest去测试，暂时先不测，这个Module在libai里面会直接应用。

**Review by xiezipeng-ML — APPROVED** [review]
